:title: About
:url:
:save_as: index.html

I am a Robertson Fellow in Biomedical Data Science at the
`Big Data Institute <https://www.bdi.ox.ac.uk/>`_, `University of
Oxford <http://www.ox.ac.uk/>`_. I develop algorithms and software
to tackle difficult problems in large-scale genomics, integrating
with the Python data science stack.

----

********
Contact
********

.. |profile_picture| image:: {static}/images/profile-picture.jpg
    :width: 200 px
    :alt: Jerome Kelleher

.. |jk_email| replace:: jerome.kelleher@bdi.ox.ac.uk
.. _jk_email: mailto:jerome.kelleher@bdi.ox.ac.uk

.. |jk_github| replace:: @jeromekelleher
.. _jk_github: https://github.com/jeromekelleher

.. |jk_twitter| replace:: @jeromekelleher
.. _jk_twitter: https://twitter.com/jeromekelleher

.. |jk_orcid| replace:: orcid.org/0000-0002-7894-5253
.. _jk_orcid: http://orcid.org/0000-0002-7894-5253

.. |jk_scholar| replace:: profile
.. _jk_scholar: https://scholar.google.co.uk/citations?user=aYTQa_AAAAAJ&hl=en&oi=ao



+---------------------+---------------------+---------------------+
| **Email**           |  |jk_email|         |                     |
+---------------------+---------------------+                     +
| **GitHub**          |  |jk_github|_       |                     |
+---------------------+---------------------+                     +
| **Twitter**         |  |jk_twitter|_      |   *Me at the BDI*   |
+---------------------+---------------------+                     +
| **Orcid**           |  |jk_orcid|_        |                     |
+---------------------+---------------------+                     +
| **Google Scholar**  |  |jk_scholar|_      | |profile_picture|   |
+---------------------+---------------------+---------------------+

**Address:**

|   Dr Jerome Kelleher
|   Big Data Institute
|   Li Ka Shing Centre for Health Information and Discovery
|   Old Road Campus
|   Oxford
|   OX3 7LF
|   UK


----

**********
Activities
**********

- I am co-organising Probabilistic Modelling in Genomics 2022 conference.
  Please see the `ProbGen22 website <https://probgen22.github.io/>`_
  for more details.

- I lead development of `tskit <https://tskit.dev/tskit/>`_, the
  tree sequence toolkit. Tskit is a Python and C API that allows us to
  work with genealogical histories at unprecedented scale.

- I also lead development of the `msprime
  <https://tskit.dev/msprime>`_ coalescent simulator, which is based on
  tskit, and can simulate of the exact coalescent with recombination over
  chromosome-sized regions with hundreds of thousands of samples.

- I'm a member of the `PopSim Consortium <https://github.com/popsim-consortium>`_
  and lead development of its first product,
  `stdpopsim <https://popsim-consortium.github.io/stdpopsim-docs/stable/index.html>`_.

- I'm part of the `Data Working Group <http://ga4gh.org/#>`_ in the `Global
  Alliance for Genomics and Health <http://genomicsandhealth.org/>`_ (GA4GH).
  I am currently participating in the design and `client side implementation
  <https://github.com/jeromekelleher/htsget>`_ of the `htsget data access protocol
  <http://samtools.github.io/hts-specs/htsget.html>`_.

- During my postdoc in theoretical population genetics , I worked primarily
  on a new model for populations evolving in a `spatial continuum
  <http://onlinelibrary.wiley.com/doi/10.1111/j.1558-5646.2010.01019.x/full>`_.
  During this time I developed two Python simulation packages: one to
  simulate a `general form <https://pypi.python.org/pypi/ercs>`_ of the model
  and another to simulate a more `specialised form
  <https://pypi.python.org/pypi/discsim>`_ much more efficiently.

- I found some bugs in `TAOCP <http://www-cs-faculty.stanford.edu/~uno/taocp.html>`__
  volume 4A while it was still in beta-test fascicle form. I am therefore a
  proud saver at the `The Bank of San Serriffe
  <http://www-cs-faculty.stanford.edu/~uno/boss.html>`__!

----

**********
CV Sketch
**********


2019-present
    Robertson Fellowship in Biomedical Data Science at the
    `Big Data Institute <https://www.bdi.ox.ac.uk/>`__,
    University of Oxford.

2014-2019
    Senior statistical programmer in the
    `McVean Group <http://www.well.ox.ac.uk/gil-mcvean>`__ at the
    `Big Data Institute <https://www.bdi.ox.ac.uk/>`__ (and formerly at the
    `Wellcome Centre for Human Genetics <http://www.well.ox.ac.uk/home>`__),
    University of Oxford.

2008-2014
    Postdoc, primarily at the `Institute of Evolutionary Biology
    <http://www.ed.ac.uk/biology/evolutionary-biology>`__, University
    of Edinburgh and also at the
    `Department of Statistics <https://www.stats.ox.ac.uk/>`__, University
    of Oxford.

2005-2008
    Seniour Software Architect, `MPSTOR <http://www.mpstor.com/>`__.

2002-2005
    PhD Computer Science,
    `University College Cork <http://www.ucc.ie/>`__
    (`thesis <http://jeromekelleher.net/downloads/k06.pdf>`__).

1998-2002
    BSc Computer Science,  `University College Cork <http://www.ucc.ie/>`__.


*************
Book Chapters
*************

**Jerome Kelleher and Konrad Lohse.**
Coalescent simulation with msprime.
Chapter 9 in *Statistical Population Genomics*, 2020.
`link <https://doi.org/10.1007/978-1-0716-0199-0_9>`__

**Jerome Kelleher and Gil McVean.**
Linkage Disequilibrium, Recombination and Haplotype Structure.
Chapter 2 in *Handbook of Statistical Genomics*, 2019.
`link <https://doi.org/10.1002/9781119487845.ch2>`__


************
Publications
************

**Dominic Nelson, Jerome Kelleher, Aaron P. Ragsdale, Claudia Moreau, Gil McVean and Simon Gravel.**
Accounting for long-range correlations in genome-wide simulations of large cohorts.
*PLOS Genetics* 16(5): e1008619, 2020.
`link <https://doi.org/10.1371/journal.pgen.1008619>`__

**Peter Ralph, Kevin Thornton and Jerome Kelleher**.
Efficiently Summarizing Relationships in Large Samples: A General Duality Between
Statistics of Genealogies and Genomes.
*Genetics*, 2020.
`link <https://doi.org/10.1534/genetics.120.303253>`__

**Jerome Kelleher, Yan Wong, Anthony W. Wohns, Chaimaa Fadil, Patrick K. Albers and Gil McVean.**
Inferring whole-genome histories in large population datasets.
*Nature Genetics*, 51:1330–1338, 2019.
`link <https://doi.org/10.1038/s41588-019-0483-y>`__

**Benjamin C. Haller,  Jared Galloway, Jerome Kelleher,  Philipp W. Messer and Peter L. Ralph.**
Tree‐sequence recording in SLiM opens new horizons for forward‐time simulation of whole genomes.
*Molecular Ecology Resources*, 19:552–566, 2019.
`link <https://doi.org/10.1111/1755-0998.12968>`__

**Jerome Kelleher, Kevin R. Thornton, Jaime Ashander and Peter L. Ralph.**
Efficient pedigree recording for fast population genetics simulation.
*PLoS Computational Biology*, 14(11): e1006581, 2018.
`link <https://doi.org/10.1371/journal.pcbi.1006581>`__

**Björn Grüning, Ryan Dale, Andreas Sjödin, Brad A. Chapman, Jillian Rowe,
Christopher H. Tomkins-Tinch, Renan Valieris, Johannes Köster and Bioconda Team.**
Bioconda: sustainable and comprehensive software distribution for the life sciences.
*Nature Methods*, 15(7):475-476, 2018.
`link <https://doi.org/10.1038/s41592-018-0046-7>`__

**Jerome Kelleher, Mike Lin,  CH Albach,  Ewan Birney,  Robert Davies,
Marina Gourtovaia, David Glazer,  Cristina Y Gonzalez,  David K Jackson,
Aaron Kemp, John Marshall, Andrew Nowak, Alexander Senf,  Jaime M Tovar-Corona,
Alexander Vikhorev, Thomas M Keane, and  GA4GH Streaming Task Team.**
htsget: a protocol for securely streaming genomic data, *Bioinformatics* 2018.
`link <https://doi.org/10.1093/bioinformatics/bty492>`__

**Jerome Kelleher, Alison M. Etheridge and Gilean McVean.**
Efficient Coalescent Simulation and Genealogical Analysis for Large Sample Sizes.
*PLoS Computational Biology*, Volume 12, May 2016, e1004842.
`link <http://dx.doi.org/10.1371/journal.pcbi.1004842>`__

**Jerome Kelleher, Alison M. Etheridge, Amandine Véber and Nicholas H. Barton.**
Spread of pedigree versus genetic ancestry in spatially distributed populations.
*Theoretical Population Biology*, Volume 108, April 2016, Pages 1–12.
`link <http://dx.doi.org/10.1016/j.tpb.2015.10.008>`__

**Jerome Kelleher, Alison M. Etheridge and Nicholas H. Barton.**
Coalescent simulation in continuous space: Algorithms for large neighbourhood size.
*Theoretical Population Biology*, Volume 95, August 2014, Pages 13–23.
`link <http://www.sciencedirect.com/science/article/pii/S0040580914000355#>`__

**Jerome Kelleher, Rob W. Ness and Daniel L. Halligan.**
Processing genome scale tabular data with wormtable.
*BMC Bioinformatics*, 14:356, 2013.
`link <http://www.biomedcentral.com/1471-2105/14/356>`__

**Jerome Kelleher, Nicholas H. Barton and Alison M. Etheridge.**
Coalescent simulation in continuous space.
*Bioinformatics*, Volume 29, Issue 7, pp. 955-956, 2013.
`link <http://bioinformatics.oxfordjournals.org/content/29/7/955.abstract>`__

**Nicholas H. Barton, Alison M. Etheridge, Jerome Kelleher and Amandine Véber.**
Inference in two dimensions: Allele frequencies versus lengths of shared sequence blocks.
*Theoretical Population Biology*, Volume 87, August 2013, Pages 105–119.
`link <http://www.sciencedirect.com/science/article/pii/S0040580913000233#>`__

**Nicholas H. Barton, Alison M. Etheridge, Jerome Kelleher and Amandine Véber.**
Genetic hitchhiking in spatially extended populations.
*Theoretical Population Biology,* Volume 87, August 2013, Pages 75–89.
`link <http://www.sciencedirect.com/science/article/pii/S0040580912001359>`__

**Nicholas H. Barton, Jerome Kelleher and Alison M. Etheridge.**
A new model for extinction and recolonisation in two dimensions: quantifying phylogeography.
*Evolution*, Volume 64, Issue 9, pp 2701-2715, 2010.
`link <http://onlinelibrary.wiley.com/doi/10.1111/j.1558-5646.2010.01019.x/full>`__

**Konrad Lohse and Jerome Kelleher**.
Measuring the degree of starshape in genealogies --- summary statistics and demographic inference.
*Genetics Research*, Volume 91, Issue 04, pp 281-292, 2009.
`link <http://dx.doi.org/10.1017/S0016672309990139>`__

**William Opperman and Jerome Kelleher**.
A Data Storage System, International Patent WO/2008/007348, 2008.
`link <http://www.wipo.int/pctdb/en/wo.jsp?wo=2008007348>`__

**Jerome Kelleher.**
Encoding Partitions as Ascending Compositions. PhD thesis, University College Cork, 2006.
`pdf <http://jeromekelleher.net/downloads/k06.pdf>`__

**Jerome Kelleher and Derek Bridge.**
An Accurate and Scalable Collaborative Recommender.
*Artificial Intelligence Review*, Volume 21, Issue 3-4, pp.193-213, 2004.
`link <http://www.springerlink.com/content/v7458560x0733q58/>`__

**Stefano Bistarelli, Jerome Kelleher and Barry O'Sullivan**.
Tradeoff Generation using Soft Constraints,
*Recent Advances in Constraints*, Springer, LNAI 3010 2004.
`link <http://www.springerlink.com/content/8t9whlq4ll27hulf/>`__

**Jerome Kelleher and Barry O'Sullivan**.
Evaluation-Based Semiring Meta-Constraints.
*Proceedings of MICAI*, Springer, LNCS 2972 Mexico, 2004.
`pdf <http://jeromekelleher.net/downloads/ko04.pdf>`__

**Jerome Kelleher and Derek Bridge**.
RecTree Centroid: An Accurate, Scalable Collaborative Recommender.
*Proceedings of AICS*, Trinity College, Dublin, pp.89-94, 2003.
`pdf <http://jeromekelleher.net/downloads/kb03.pdf>`__

**Stefano Bistarelli, Jerome Kelleher and Barry O'Sullivan.**
Symmetry Breaking in Soft CSPs.
*Proceedings of AI-2003*, Springer, Cambridge, UK, 2003.
`pdf <http://jeromekelleher.net/downloads/bko03.pdf>`__

**Jerome Kelleher and Barry O'Sullivan.**
Evaluation-Based Semiring Meta-Constraints.
*Proceedings of AICS-2003*, Poster Paper, Dublin, Ireland, 2003.

**Jerome Kelleher and Barry O'Sullivan**.
Optimising the Representation and Evaluation of Semiring Combination Constraints.
Principles and Practice of Constraint Programming - CP2003, LNCS, 2003.

**Derek Bridge and Jerome Kelleher**.
Experiments in Sparsity reduction: Using Clustering in Collaborative Recommenders.
*Proceedings of AICS-2002*, LNAI 2464, Springer, pp.144-149, 2002.
`pdf <http://jeromekelleher.net/downloads/bk02.pdf>`__

+++++++++
Preprints
+++++++++

**Jerome Kelleher**.
The large-parts formula for p(n).
*arXiv*, 1002.1458, 2010
`link <http://arxiv.org/abs/1002.1458>`__

**Jerome Kelleher and Barry O'Sullivan**.
Generating All Partitions: A Comparison Of Two Encodings.
*arXiv*, 0909.2331, 2009
`link <http://arxiv.org/abs/0909.2331>`__

